### DISCUSSION TITLE
The column `new` does not exist in the current database on prisma create

### DISCUSSION THREAD

**Discussion by kvnam** [prisma/prisma#13423] [Q&A] [answered]
Hello,

I'm a little stumped with this error. All I did was change the name of a column (in Supabase) and then run `npx prisma db pull` and `npx prisma generate` in my Node express backend. Now when I try to create an entry I get this error:

`The column "new" does not exist in the current database` on prisma create

There is no column by that name anywhere in the DB, and I logged to see that the input passed to create as `data` is also clean and contains no `new` property. I re-ran the prisma commands a few more times in case the prisma schema was corrupt somehow, but that did not help either.

Any idea what could be causing this?

```{
     "code": P2022", 
      clientVersion: 3.12.0, 
      meta: { column:new }
    }
      data:null
    }
   }
```
Update: I tried deleting and re-adding the column, updating the schema for each instance, and it gave an error in both cases (column removed and column added back)

Any help would be greatly appreciated, thanks!

**Comment by alucardu** [comment]
Do you need to run the migration?

**Reply by kvnam** [reply]
I'm not using prisma migrate at all, Prisma is not managing my DB, it's the other way around only in my case. So I update the Prisma schema after making changes to  the DB with prisma db pull.

**Comment by janpio** [comment, marked as answer]
All you did was rename a column from `foo` to `bar` (via external SQL changes, then ran `prisma db pull`) and now a `create` in Prisma complains about another column name `telephone` not existing? Can you paste the full output you are getting please?

**Reply by kvnam** [reply]
@janpio - Sure this is the error I get

```
PrismaClientKnownRequestError:
Invalid `prisma.partners.create()` invocation:


  The column `new` does not exist in the current database.
    at Object.request (/node_modules/@prisma/client/runtime/index.js:45578:15)
    at PrismaClient._request (/node_modules/@prisma/client/runtime/index.js:46405:18)
    at Object.createPartner (/.webpack/service/src/webpack:/@src/controllers/partners/api.js:50:12)
    at Object.createPartner (.webpack/service/src/webpack:/@/src/controllers/partners/resolvers.js:70:22) {
  code: 'P2022',
  clientVersion: '3.12.0',
  meta: { column: 'new' }
}
```

The input sent is

```
addressLine1: "3200 South Las Vegas Boulevard"
city: "Las Vegas"
companyName: "Test Partner"
email: "testpartner@email.com"
officePhone: "4327463"
state: "Nevada"
status: "active"
zipCode: "89109"
latitude: "-23.4324"
longitude: "34.432432"  // This was the renamed column (went from longtude to longitude (typo fix)
```

The input does not contain `new` I logged it to check. In fact the entire database never had column named `new`

**Reply by janpio** [reply]
That is _weird_. Try https://pris.ly/d/debugging and https://pris.ly/d/logging please to get some more information on what is happening under the hood. Somewhere this `new` needs to be added along the way.

**Reply by kvnam** [reply]
Sure @janpio I enabled client debugging and I can see `new` does not exist in the generatedRequest
```
 prisma:client Generated request: +0ms
  prisma:client mutation {
  prisma:client   createOnepartners(data: {
  prisma:client     status: "active"
  prisma:client     email: "testpartner@email.com"
  prisma:client     officePhone: "4327463"
  prisma:client     addressLine1: "3200 South Las Vegas Boulevard"
  prisma:client     city: "Las Vegas"
  prisma:client     state: "Nevada"
  prisma:client     zipCode: "89109"
  prisma:client     companyName: "Test Partner"
  prisma:client     latitude: "36.1276769"
  prisma:client     longitude: "-115.1685847"
  prisma:client     partners_contacts: {
  prisma:client       create: {
  prisma:client         contacts: {
  prisma:client           connectOrCreate: {
  prisma:client             where: {
  prisma:client               id: 1
  prisma:client             }
  prisma:client             create: {
  prisma:client
  prisma:client             }
  prisma:client           }
  prisma:client         }
  prisma:client       }
  prisma:client     }
  prisma:client   })
```
No new in the query either:
```
prisma:query BEGIN
prisma:query DEALLOCATE ALL
prisma:query INSERT INTO "public"."partners" ("city","status","email","address_line1","zipcode","officePhone","companyName","state","latitude","isActive","longitude") VALUES ($1,$2,$3,$4,$5,$6,$7,$8,$9,$10,$11) RETURNING "public"."partners"."id"
prisma:query ROLLBACK
```

I then added logs at `node_modules/@prisma/client/runtime/index.js:45578:15` which led to `PrismaClient._request` where I logged `internalParams` all correct.

```
 args: {
    data: {
      status: 'active',
      email: 'testpartner@email.com',
      officePhone: '4327463',
      addressLine1: '3200 South Las Vegas Boulevard',
      city: 'Las Vegas',
      state: 'Nevada',
      zipCode: '89109',
      companyName: 'Test Partner',
      latitude: '36.1276769',
      longitude: '-115.1685847',
      partners_contacts: [Object]
    },
    include: {
      my_includes: [Object],
    }
  },
  dataPath: [],
  action: 'create',
  model: 'partners',
  clientMethod: 'partners.create',
  runInTransaction: false,
  transactionId: undefined,
  lock: undefined,
  callsite: 'Error: \n' +
    '    at getCallSite (/node_modules/@prisma/client/runtime/index.js:42832:10)\n' +
    '    at Proxy.create (/node_modules/@prisma/client/runtime/index.js:42986:22)\n' +
```
`/node_modules/@prisma/client/runtime/index.js:42986:22`  ends at `getCallSite` function in `applyFluent`. I'll keep adding more info, but if any of this indicates what could be wrong, please do let me know. Thanks!

**Reply by janpio** [reply]
Nothing pops out to me as well.

What does look a bit weird is the `partners_contacts` object in the mutation. `create` being empty there is a bit unexpected. What does your full Prisma query look like?

**Reply by janpio** [reply]
Ah, ignore that - I just noticed that it is already rolling back after the `INSERT` that does look just fine on paper. Use https://www.prisma.io/docs/concepts/components/prisma-client/working-with-prismaclient/logging#event-based-logging to get the params it is trying to execute in that query - maybe something there is funky.

**Reply by kvnam** [reply]
Nothing strange there either..... 

```
Query: BEGIN
Params: []
Duration: 22ms
Query: DEALLOCATE ALL
Params: []
Duration: 26ms
Query: INSERT INTO "public"."partners" ("city","status","email","address_line1","zipcode","officePhone","companyName","state","latitude","isActive","longitude") VALUES ($1,$2,$3,$4,$5,$6,$7,$8,$9,$10,$11) RETURNING "public"."partners"."id"
Params: ["Las Vegas","active","testpartner@email.com","3200 South Las Vegas Boulevard","89109","4327463","Test Partner","Nevada","36.1276769",true,"-115.1685847"]
Duration: 50ms
Query: ROLLBACK
Params: []
Duration: 23ms
```

**Reply by kvnam** [reply]
@janpio - I found the cause, :confounded: it was a trigger, complaining about a column that was deleted from the table, but the trigger was not removed. The `new` was the `NEW` record in the trigger. I am very sorry for wasting your time with this, thank you so much for looking into it, I appreciate it.

**Reply by janpio** [reply]
No time wasted here, that was super interesting and I am very happy you found the cause. It is interesting how the database drops that info `NEW` on you with no context.

**Reply by kvnam** [reply]
Yep, the error message is not very helpful. The database (supabase) returned `record "new" has no field "fullName"` and Prisma returned this as `The column `new` does not exist in the current database.` Supabase has not modified the error I believe, is it because Prisma does not map to Postgres trigger function errors formats..?

**Reply by janpio** [reply]
Ah, that is super valuable information!

Yes, I expect Prisma does not know this and somehow creates a misleading error message. Can you please open a bug report issue for this? Optimally with a reproduction of how to get to this? Then we could reproduce and schedule fixing this - to hopefully not make more people waste their time in the future on it.

**Reply by kvnam** [reply]
Sure, I'll raise a bug report issue for this, glad I can help 👍

**Reply by snikch** [reply]
Just found this via Google. Doesn't look like this was ever fixed? Exact same issue - supabase trigger.

**Reply by cesarvspr** [reply]
here a trigger using a column that was dropped was causing this

**Reply by hughgardiner** [reply]
> @janpio - I found the cause, 😖 it was a trigger, complaining about a column that was deleted from the table, but the trigger was not removed. The `new` was the `NEW` record in the trigger. I am very sorry for wasting your time with this, thank you so much for looking into it, I appreciate it.

this should be the *answer*, saved me so much headache

**Reply by titusdecali** [reply]
Thanks, a trigger also caused this problem for me after I temporarily thought I needed an edge function. After adding a /functions folder to my project I believe it generated the trigger automatically.

**Reply by MrChrisRodriguez** [reply]
fwiw - thanks for bringing this up. This is still happening to folks (like me) and the error message is a red herring, so this as top google result is our saving grace!
